pkg/util/phone/phone.go

```diff
@@ -41,6 +51,29 @@ func Parse(nationalNumber string, callingCode string) (e164 string, err error) {
 	return
 }
 
+func ParseWithValidation(nationalNumber string, callingCode string) (e164 string, err error) {
```

Comment:
I think we do not need both Parse and ParseWithValidation. Make Parse behave like ParseWithValidation. Parse was always meant to return validated phone number in E.164 format.

Reply:
Can you also add test cases to verify if the sibling function EnsureE164 accepts invalid input like `+85222` or `+85222a`? Please help fix it if it is the case.

Reply:
BTW, the two commits should be a single commit. It is more logical to have the new test cases committed along with the changed behavior.

Reply:
I expect +85222 passes EnsureE164 as it can be parsed, and formatted correctly, +85222a should fail the test, as even it can be parsed, formatting from uint64 national number gives number, hence not equal to original. You want EnsureE164 to do phone number validation as well? This means phone number like +85212345678 will fail the check as well, as +852 number do not begins with 1 for example.

Reply:
Yes. For our use case, we never accept invalid phone number. We use phone numbers to deliver SMS message so they must be valid.

Reply:
OK, I will call `isValidNumber` in `EnsureE164`